### DISCUSSION TITLE
Procedural Asteroids SPS

### DISCUSSION THREAD

**Discussion by jelster** [jelster/space-truckers#25] [Ideas]
The `createAsteroids` function in [this snippet](https://playground.babylonjs.com/#9EI3H4%2310) shows generating an asteroid belt using the solid particle system and some of the Scalar Random generation functions. The latest revision of that snippet is 17 and shows PBR Material for the asteroids, although it also refers to a number of local host assets as well, so it would be worth re-implementing PBR stuff into the online-only snippet at some point

**Comment by eldinor** [comment]
I went to the Dark Space :)
![image](https://user-images.githubusercontent.com/25833327/106370017-7c1d2c00-6367-11eb-9a74-c140e371d775.png)

**Reply by jelster** [reply]
Ha something missing from this is a simple mesh intersect action to detect when the box reaches the skybox, makes for some deep space adventures 🤓🧐😎

**Reply by eldinor** [reply]
I still couldn't achieve proper launch - seems that there should be the tube visible?

**Reply by jelster** [reply]
Which snippet revision was that? Iirc, the ones above 10 use localhost assets,[this revision number 10](https://playground.babylonjs.com/#9EI3H4%2310) creates the tube on successfully reaching the destination![image](https://user-images.githubusercontent.com/1508500/106399794-74e63480-63e0-11eb-8e14-6d83fc746806.jpeg)

**Reply by eldinor** [reply]
Seems this is a tricky part of the game.
Spent five minutes before my first tube
![image](https://user-images.githubusercontent.com/25833327/106399916-8bb27880-642c-11eb-932f-dbed8123cabc.png)

**Reply by jelster** [reply]
I think you've brought up something about this I was taking for granted - if you put launch power to max and point the launcher between the sun and planet directly towards the destination, the route is consistent, easy, and short. 

As you pointed out, it took a while for you to get your first success, do you think it would spoil the discovery aspect of this if the game gave you a hint for the above easy path?

**Reply by eldinor** [reply]
Well, inspite I have seen as you did it and even after your explanations it is not so evident and easy to find the tube (taking into the account that usually I am quite good with games).
Maybe we'll need a kind of short tutorial video and definitely some textual guide and tips.

**Reply by jelster** [reply]
Thinking on this a bit more, given the difficulty of the task, do players need more of a reward for success? Can definitely imagine some sound effects and visuals that could bring that gratification

**Reply by eldinor** [reply]
Well, any hints and tips here will be useful for players.
Probably it is also good idea to simplify the task (or suggest automatic choice for beginners) since the main fun will be in driving.

**Reply by jelster** [reply]
I think we could accomplish this easily by changing the starting position and values for the launcher so that just pressing launch without changing anything will have a high likelihood of making it to the destination.

Explanatory text could label it the beginners or the easy route, showing the lack of difficulty in the route translates to much lower maximum possible score for the trip
